File: dotnet-nunit-selenium/Pages/FormPageSite45.cs
using OpenQA.Selenium;
using Selenium.WebDriver.SelfHealing;

namespace SelfHealingLocatorsDemo;

class FormPageSite45 : BasePage
{
    By _buttonLocator;
    By _linkLocator = By.Id("link");

    public FormPageSite45(string url, By buttonLocator, string classification, ISelfHealingWebDriver webDriver) 
    : base(webDriver)
    {
        this.WebDriver.Navigate().GoToUrl(url);
        _buttonLocator = buttonLocator;
        Classification = classification;
    }

    public void RevealHiddenLink() => WebDriver.FindElement(_buttonLocator).Click();

    public string GetPageLink() => WebDriver.FindElement(_linkLocator).Text;
}

File: java-testng-selenium/src/test/java/selfhealinglocators/Pages/FormPageSite45.java
package selfhealinglocators.Pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class FormPageSite45 extends PageBase
{
    By linkLocator = By.id("link");
    By buttonLocator; 

    public FormPageSite45(String url, By buttonLocator, WebDriver webDriver){
        super(webDriver);
        webDriver.navigate().to(url);
        this.buttonLocator = buttonLocator;
    }

    public void revealHiddenLink() {
        webDriver.findElement(buttonLocator).click();
    }

    public String getPageLink() {
        return webDriver.findElement(linkLocator).getText();
    }
}
